plan/aggregation_push_down.go

```diff
@@ -391,6 +391,16 @@ func (a *aggregationOptimizer) aggPushDown(p LogicalPlan) LogicalPlan {
 				}
 				union.SetChildren(newChildren...)
 				union.SetSchema(pushedAgg.schema)
+			} else if ds, ok1 := child.(*DataSource); ok1 {
```

Comment:
Can we use the condition of `GenValues != nil` to reduce the possibility of traversal?